### PR TITLE
Expand docs of multi-address behavior of some UDP/TCP APIs.

### PR DESCRIPTION
**Pull request by frewsxcv**
Fixes https://github.com/rust-lang/rust/issues/22569.

### CODE REVIEW

File: src/libstd/net/tcp.rs [hunk 1]
```diff
@@ -129,6 +132,23 @@ impl TcpStream {
     ///     println!("Couldn't connect to server...");
     /// }
     /// ```
+    ///
+    /// Open a TCP connection to `127.0.0.1:8080`. If the connection fails, open
+    /// a TCP connection to `127.0.0.1:8081`:
+    ///
+    /// ```no_run
+    /// use std::net::TcpStream;
```
Comment (kennytm) [inline src/libstd/net/tcp.rs]: The doc tests failed, `SocketAddr` needs to be imported first.

<details>

```
[00:57:39] failures:
[00:57:39] 
[00:57:39] ---- net/tcp.rs - net::tcp::TcpListener::bind (line 602) stdout ----
[00:57:39] 	error[E0433]: failed to resolve. Use of undeclared type or module `SocketAddr`
[00:57:39]  --> net/tcp.rs:7:5
[00:57:39]   |
[00:57:39] 7 |     SocketAddr::from(([127, 0, 0, 1], 80)),
[00:57:39]   |     ^^^^^^^^^^ Use of undeclared type or module `SocketAddr`
[00:57:39] 
[00:57:39] error[E0433]: failed to resolve. Use of undeclared type or module `SocketAddr`
[00:57:39]  --> net/tcp.rs:8:5
[00:57:39]   |
[00:57:39] 8 |     SocketAddr::from(([127, 0, 0, 1], 443)),
[00:57:39]   |     ^^^^^^^^^^ Use of undeclared type or module `SocketAddr`
[00:57:39] 
[00:57:39] thread 'rustc' panicked at 'couldn't compile the test', /checkout/src/librustdoc/test.rs:280:12
[00:57:39] note: Run with `RUST_BACKTRACE=1` for a backtrace.
[00:57:39] 
[00:57:39] ---- net/tcp.rs - net::tcp::TcpStream::connect (line 140) stdout ----
[00:57:39] 	error[E0433]: failed to resolve. Use of undeclared type or module `SocketAddr`
[00:57:39]  --> net/tcp.rs:7:5
[00:57:39]   |
[00:57:39] 7 |     SocketAddr::from(([127, 0, 0, 1], 8080)),
[00:57:39]   |     ^^^^^^^^^^ Use of undeclared type or module `SocketAddr`
[00:57:39] 
[00:57:39] error[E0433]: failed to resolve. Use of undeclared type or module `SocketAddr`
[00:57:39]  --> net/tcp.rs:8:5
[00:57:39]   |
[00:57:39] 8 |     SocketAddr::from(([127, 0, 0, 1], 8081)),
[00:57:39]   |     ^^^^^^^^^^ Use of undeclared type or module `SocketAddr`
[00:57:39] 
[00:57:39] thread 'rustc' panicked at 'couldn't compile the test', /checkout/src/librustdoc/test.rs:280:12
[00:57:39] 
[00:57:39] 
[00:57:39] failures:
[00:57:39]     net/tcp.rs - net::tcp::TcpListener::bind (line 602)
[00:57:39]     net/tcp.rs - net::tcp::TcpStream::connect (line 140)
```

</details>

File: src/libstd/net/tcp.rs [hunk 2]
```diff
@@ -557,16 +577,36 @@ impl TcpListener {
...
     /// use std::net::TcpListener;
     ///
     /// let listener = TcpListener::bind("127.0.0.1:80").unwrap();
     /// ```
+    ///
+    /// Create a TCP listener bound to `127.0.0.1:80`. If that fails, create a
+    /// TCP listener bound to `127.0.0.1:443`:
+    ///
+    /// ```no_run
+    /// use std::net::TcpListener;
```
Comment (kennytm) [inline src/libstd/net/tcp.rs]: ditto.

**Comment by frewsxcv** [comment]
Imports fixed in the latest force push.

**Comment by frewsxcv** [comment]
fixed another import issue

@bors r=alexcrichton

```diff
diff --git a/src/libstd/net/tcp.rs b/src/libstd/net/tcp.rs
index 2fb5d2b0ac..5467eff202 100644
--- a/src/libstd/net/tcp.rs
+++ b/src/libstd/net/tcp.rs
@@ -599,7 +599,7 @@ impl TcpListener {
     /// TCP listener bound to `127.0.0.1:443`:
     ///
     /// ```no_run
-    /// use std::net::{SocketAddr, TcpStream};
+    /// use std::net::{SocketAddr, TcpListener};
     ///
     /// let addrs = [
     ///     SocketAddr::from(([127, 0, 0, 1], 80)),
```

**Pull request merged**